File: perl/v2_translate.pl
#!/usr/bin/perl
# v2_translate.pl
# Instalar: cpan SOAP::Lite Lingua::Translate
# Ejecutar: perl v2_translate.pl 10

use strict;
use warnings;
use SOAP::Lite;
use Lingua::Translate;

my $num    = $ARGV[0] // 10;
my $client = SOAP::Lite->service(
    'https://www.dataaccess.com/webservicesserver/NumberConversion.wso?WSDL');

my $en = $client->NumberToWords($num);

my $xl = Lingua::Translate->new(src => 'en', dest => 'es');
print $xl->translate($en), "\n";

File: nodejs/v2_translate.js
// v2_translate.js
// Ejecutar: node v2_translate.js 10

const soap = require('soap');

const WSDL = 'https://www.dataaccess.com/webservicesserver/NumberConversion.wso?WSDL';
const num  = parseInt(process.argv[2]) || 10;

async function main() {
    const { translate } = await import('@vitalets/google-translate-api');

    soap.createClient(WSDL, async (err, client) => {
        if (err) throw err;
        const [result] = await client.NumberToWordsAsync({ ubiNum: num });
        const en = result.NumberToWordsResult;
        const res = await translate(en, { from: 'en', to: 'es' });
        console.log(res.text);
    });
}

main();